src/lib.rs

```diff
@@ -17,3 +24,25 @@ pub enum Error {
     #[error("An error occured in the backend: {0}")]
     Backend(#[from] backend::Error),
 }
+
+pub fn init_logging(level: &LogLevel, path: &PathBuf) {
+    let log_file = File::create(path).unwrap();
+
+    let subscriber_builder = tracing_subscriber::fmt()
+        .with_writer(log_file)
+        .with_max_level(LevelFilter::from(level))
+        .without_time()
+        .with_ansi(true)
+        .with_target(false)
+        .with_file(true);
+
+    if std::env::var_os("RUST_LOG").is_some() {
+        subscriber_builder
+            .with_env_filter(EnvFilter::from_default_env())
+            .init();
+    } else {
+        subscriber_builder.init();
+    }
```

Comment:
That's not possible, because [`with_env_filter`](https://docs.rs/tracing-subscriber/0.3.18/tracing_subscriber/fmt/struct.SubscriberBuilder.html#method.with_env_filter) is consuming `self` and returning the modified subscriber.